ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -80,21 +80,53 @@ def main(argv=sys.argv[1:]):
     parser.add_argument(
         '--xunit-file',
         help='Generate a xunit compliant XML file')
+
+    parser.add_argument(
+        '--filter-repo-path',
+        help='Used to test only packages from one repo. '
+        'Accepts a path to repo containing ros packages.  If this path contains '
+        'a .clang-tidy config file, the tests will use that config.')
+    parser.add_argument(
+        '--filter-diff',
+        help='Used to only run tests on packages with changes since a specified git commit. '
+        'Accepts branch, tag, or git commit hash to generate diff.  Requires --filter-repo-path.')
+
+    parser.add_argument(
+        '--verbose',
+        action='store_true',
+        help='Show clang-tidy commands as they are run.')
     args = parser.parse_args(argv)
 
+    if args.filter_repo_path:
+        repo_clang_tidy = args.filter_repo_path + '/.clang-tidy'
+        if os.path.exists(repo_clang_tidy):
+            args.config_file = repo_clang_tidy
+
```

Comment:
This sort of default behavior I could see being an issue if someone specified a --config option and it was different from one found in the root of the repo they wanted to filter on.  Lmk what you think.  I'm not really attached to this feature and would happily remove it.

Reply:
Yeah, I think it would be confusing if specifying `--filter-repo-path` would change the config that is used.

If I recall correctly, currently the [globally common .clang-tidy config](https://github.com/ament/ament_lint/blob/master/ament_clang_tidy/ament_clang_tidy/configuration/.clang-tidy) would be used (unless `--config` is specified) regardless of the presence of a local `.clang-tidy`.

We could change it so that a local `.clang-tidy` would always take precedence, regardless of whether or not `--filter-repo-path` was specified.